src/DependencyInjection/Compiler/SearchQueryMappersPass.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace EzSystems\EzPlatformGraphQL\DependencyInjection\Compiler;
+
+use EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\SearchQueryMapper;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class SearchQueryMappersPass implements CompilerPassInterface
+{
+    const ID = 'EzSystems\EzPlatformGraphQL\GraphQL\InputMapper\SearchQueryMapper';
+
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->has(self::ID)) {
+            return;
+        }
+
+        $definition = $container->findDefinition(SearchQueryMapper::class);
+        $taggedServices = $container->findTaggedServiceIds('ezplatform_graphql.query_input_visitor');
+
+        $queryInputVisitors = [];
+        foreach ($taggedServices as $id => $tags) {
+            foreach ($tags as $tag) {
+                if (isset($tag['inputKey'])) {
+                    $queryInputVisitors[$tag['inputKey']] = new Reference($id);
```

Comment:
Note: the advantage of this is that a query input visitor can be overridden using the key. Useful.